x-pack/legacy/plugins/reporting/server/core.ts

```diff
@@ -5,33 +5,41 @@
  */
 
 import * as Rx from 'rxjs';
-import { first, mapTo } from 'rxjs/operators';
+import { first, mapTo, map } from 'rxjs/operators';
 import {
   ElasticsearchServiceSetup,
   KibanaRequest,
-  SavedObjectsClient,
   SavedObjectsServiceStart,
   UiSettingsServiceStart,
+  IRouter,
+  SavedObjectsClientContract,
+  BasePath,
 } from 'src/core/server';
+import { SecurityPluginSetup } from '../../../../plugins/security/server';
 import { ReportingPluginSpecOptions } from '../';
 // @ts-ignore no module definition
 import { mirrorPluginStatus } from '../../../server/lib/mirror_plugin_status';
+import { LicensingPluginSetup } from '../../../../plugins/licensing/server';
 import { XPackMainPlugin } from '../../xpack_main/server/xpack_main';
-import { PLUGIN_ID } from '../common/constants';
 import { screenshotsObservableFactory } from '../export_types/common/lib/screenshots';
-import { ServerFacade } from '../server/types';
+import { ServerFacade, ScreenshotsObservableFn } from '../server/types';
 import { ReportingConfig } from './';
 import { HeadlessChromiumDriverFactory } from './browsers/chromium/driver_factory';
-import { checkLicenseFactory, getExportTypesRegistry, LevelLogger } from './lib';
+import { checkLicense, getExportTypesRegistry, LevelLogger } from './lib';
 import { ESQueueInstance } from './lib/create_queue';
 import { EnqueueJobFn } from './lib/enqueue_job';
 import { registerRoutes } from './routes';
-import { ReportingSetupDeps } from './types';
 
-interface ReportingInternalSetup {
+export interface ReportingInternalSetup {
```

Comment:
Looks like this is just exported for tests to be able to mock it: `{ blah } as unknown as ReportingInternalSetup` whereas I think the mocks can just do: `{ blah } as any` with the same result. So, in favor of not exporting something unless it's essential, this shouldn't be exported. 

```suggestion
interface ReportingInternalSetup {
```

Reply:
On second thought, this seems like it is needed for `createMockReportingCore`. So, nevermind this comment!